core/server/models/session.js

```diff
@@ -0,0 +1,78 @@
+const ghostBookshelf = require('./base');
+
+const Session = ghostBookshelf.Model.extend({
+    tableName: 'sessions',
+
+    parse: function parse(attrs) {
+        attrs.session_data = JSON.parse(attrs.session_data);
+        return attrs;
+    },
+
+    format: function format(attrs) {
+        if (attrs.session_data) {
+            attrs.session_data = JSON.stringify(attrs.session_data);
+        }
+        return attrs;
+    },
+
+    user: function () {
+        return this.belongsTo('User');
+    }
+}, {
+
+    permittedOptions(methodName) {
+        const permittedOptions = ghostBookshelf.Model.permittedOptions.call(this, methodName);
+        if (methodName === 'upsert') {
+            return permittedOptions.concat('session_id');
+        }
+        if (methodName === 'destroy') {
+            return permittedOptions.concat('session_id');
+        }
+        return permittedOptions;
+    },
+
+    destroy(unfilteredOptions) {
+        if (unfilteredOptions.id) {
+            return ghostBookshelf.Model.destroy.call(this, unfilteredOptions);
```

Comment:
I think this code can get rid of the manual implementation of manually destroying the session. I have not tested this.

```
if (options.session_id) {
  this.data = {session_id: options.session_id};
}

return ghostBookshelf.Model.destroy.call(this, unfilteredOptions);
```

base destroy
```
const data = this.data || {id: options.id}
return this.forge(data)...
```

---

We should in general improve the destroy method to allow passing `destroy(data, options)` - but not part of your task.

Reply:
> We should in general improve the destroy method to allow passing destroy(data, options) - but not part of your task 

I agree with this - would be nice to clean it all up a bit :)

Though currently the base destroy manually passes `{id: options.id}` so I can't get rid of the manual implementation here without updating base destroy.

Reply:
Yeah, separate PR :)